internal/app/agent/register/register.go

```diff
@@ -0,0 +1,50 @@
+package register
+
+import (
+	"strings"
+
+	nats "github.com/nats-io/go-nats"
+)
+
+const (
+	HEARTBEAT = "heartbeat.*"
+)
+
+var _ Service = (*register)(nil)
+
+type Service interface {
+	Applications() map[string]*Application
```

Comment:
Please be consistent in naming and terminology. There is no need to mix `applications` and `services`.

Reply:
service here is register service that keeps track of applications registred via heartbeat

Reply:
Got it. What I find confusing is Register service. Do we need it? Or we can simply put `Applications` method directly in our Service?

Reply:
We do not need a new service for this, this should be a part of Agent main service.